base/commands/viridian/viridian_it_test.go

```diff
@@ -0,0 +1,130 @@
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	_ "github.com/hazelcast/hazelcast-commandline-client/base/commands"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/it"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/viridian"
+)
+
+func TestViridian(t *testing.T) {
+	it.MarkViridian(t)
+	defer func() {
+		// make sure Viridian clusters are deleted
+		t.Logf("TestViridian cleanup...")
+		tcx := it.TestContext{T: t}
+		tcx.Tester(func(tcx it.TestContext) {
+			ctx := context.Background()
+			infos := check.MustValue(tcx.Viridian.ListClusters(ctx))
+			for _, info := range infos {
+				if err := tcx.Viridian.DeleteCluster(ctx, info.ID); err != nil {
+					tcx.T.Logf("ERROR while cleaning up cluster: %s", err.Error())
+				}
+
+			}
+		})
+	}()
+	testCases := []struct {
+		name string
+		f    func(t *testing.T)
+	}{
+		{"listClusters", listClustersTest},
+		{"loginWithEnvVariables", loginWithEnvVariablesTest},
+		{"loginWithParams", loginWithParamsTest},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, tc.f)
+	}
+}
+
+func loginWithParamsTest(t *testing.T) {
+	tcx := it.TestContext{T: t}
+	tcx.Tester(func(tcx it.TestContext) {
+		ctx := context.Background()
+		tcx.CLCExecute(ctx, "viridian", "login", "--api-key", it.ViridianAPIKey(), "--api-secret", it.ViridianAPISecret())
+		tcx.AssertStdoutContains("Viridian token was fetched and saved.")
+	})
+}
+
+func loginWithEnvVariablesTest(t *testing.T) {
+	tcx := it.TestContext{T: t}
+	tcx.Tester(func(tcx it.TestContext) {
+		ctx := context.Background()
+		it.WithEnv(viridian.EnvAPIKey, it.ViridianAPIKey(), func() {
+			it.WithEnv(viridian.EnvAPISecret, it.ViridianAPISecret(), func() {
+				tcx.CLCExecute(ctx, "viridian", "login")
+				tcx.AssertStdoutContains("Viridian token was fetched and saved.")
+			})
+		})
+	})
+}
+
+func listClustersTest(t *testing.T) {
+	viridianTester(t, func(ctx context.Context, tcx it.TestContext) {
+		tcx.CLCExecute(ctx, "viridian", "list-clusters")
+		tcx.AssertStderrContains("OK")
+		cid := ensureClusterRunning(ctx, tcx)
+		tcx.CLCExecute(ctx, "viridian", "list-clusters")
+		tcx.AssertStderrContains("OK")
+		tcx.AssertStdoutContains(cid)
+
+	})
+}
+
+func viridianTester(t *testing.T, f func(ctx context.Context, tcx it.TestContext)) {
+	tcx := it.TestContext{T: t}
+	tcx.Tester(func(tcx it.TestContext) {
+		ctx := context.Background()
+		tcx.CLCExecute(ctx, "viridian", "login", "--api-key", it.ViridianAPIKey(), "--api-secret", it.ViridianAPISecret())
+		tcx.AssertStdoutContains("Viridian token was fetched and saved.")
+		tcx.WithReset(func() {
+			f(ctx, tcx)
+		})
+	})
+}
+
+func ensureClusterRunning(ctx context.Context, tcx it.TestContext) string {
+	info, err := tcx.Viridian.CreateCluster(ctx, it.UniqueClusterName())
+	if err != nil {
+		tcx.T.Logf("Ignoring error: %s", err.Error())
+		infos := check.MustValue(tcx.Viridian.ListClusters(ctx))
+		info = infos[0]
```

Comment:
If slice is empty this could panic, is this ok?

Reply:
Yes, in the common case the error should be about "adding payment method" since there would already be a running cluster.